tar/multitape/multitape_write.c

```diff
@@ -423,8 +423,8 @@ writetape_open(uint64_t machinenum, const char * cachedir,
 	if ((d->dryrun == 0) && multitape_cleanstate(cachedir, machinenum, 0))
 		goto err4;
 
-	/* Get sequence number. */
-	if (multitape_sequence(cachedir, lastseq))
+	/* If this isn't a dry run, get the sequence number. */
+	if ((d->dryrun == 0) && (multitape_sequence(cachedir, lastseq)))
```

Comment:
If we are doing a dryrun, this does leave `lastseq` unset and it is passed into `storage_write_start`.

However, as long as we are doing a dryrun, `storage_write_start` does not look at `lastseq`.  (code and comment changed below).

If you'd like me to add

    memset(lastseq, 0, 32);

here, I have no objection, but it's not required given the current state of the code.  (and hopefully nobody will be changing `storage_write_start` in the future in a manner which require `lastseq`, given the comment I added to its "contract".